prover/proof_producer/guardian_producer.go

```diff
@@ -5,17 +5,20 @@ import (
 	"math/big"
 
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/log"
 
 	"github.com/taikoxyz/taiko-client/bindings"
 	"github.com/taikoxyz/taiko-client/bindings/encoding"
 )
 
 // GuardianProofProducer always returns an optimistic (dummy) proof.
-type GuardianProofProducer struct{ *DummyProofProducer }
+type GuardianProofProducer struct {
+	LivenessBond bool
```

Comment:
IMO this should be local variable and named better, ie : `returnLivenessBond: bool`, and we should use the common Go pattern of `New` to initialize the guardian proof producer, ie:

```g
func NewGuardianProofProducer(returnLivenessBond bool) (*GuardianProofProducer) {
  gp := &GuardianProofProducer {
      returnLivenessBond: returnLivenessBond,
  }
  
  // we only need to intantiatea dummy proof producer if we are not returning fake liveness bonds
  if !returnLivenessBond {
      gp.dummyProofProducer = new(DummyProofProducer),
  }
  
  return gp
}
```